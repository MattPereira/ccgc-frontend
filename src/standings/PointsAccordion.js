import React from "react";
import { Accordion, Table } from "react-bootstrap";

/**
 * Points Accordion with tables inside.
 */

const PointsAccordion = () => {
  return (
    <Accordion className="text-center">
      <Accordion.Item eventKey="0">
        <Accordion.Header>Strokes</Accordion.Header>
        <Accordion.Body>
          <div className="row justify-content-center">
            <div className="col-md-6">
              <Table bordered striped variant="dark">
                <thead>
                  <tr>
                    <th>Position</th>
                    <th>Points</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>1st</td>
                    <td>25</td>
                  </tr>
                  <tr>
                    <td>2nd</td>
                    <td>20</td>
                  </tr>
                  <tr>
                    <td>3rd</td>
                    <td>15</td>
                  </tr>
                  <tr>
                    <td>4th</td>
                    <td>10</td>
                  </tr>
                  <tr>
                    <td>5th</td>
                    <td>5</td>
                  </tr>
                </tbody>
              </Table>
            </div>
          </div>
        </Accordion.Body>
      </Accordion.Item>
      <Accordion.Item eventKey="1">
        <Accordion.Header>Putts</Accordion.Header>
        <Accordion.Body>
          <div className="row justify-content-center">
            <div className="col-md-6">
              <Table bordered variant="dark" striped>
                <thead>
                  <tr>
                    <th>Position</th>
                    <th>Points</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>1st</td>
                    <td>6</td>
                  </tr>
                  <tr>
                    <td>2nd</td>
                    <td>4</td>
                  </tr>
                  <tr>
                    <td>3rd</td>
                    <td>2</td>
                  </tr>
                </tbody>
              </Table>
            </div>
          </div>
        </Accordion.Body>
      </Accordion.Item>
      <Accordion.Item eventKey="2">
        <Accordion.Header>Greenies</Accordion.Header>
        <Accordion.Body>
          <div className="row justify-content-center">
            <div className="col-md-6">
              <Table bordered variant="dark" striped>
                <thead>
                  <tr>
                    <th>Event</th>
                    <th>Points</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>Greenie</td>
                    <td>1</td>
                  </tr>
                  <tr>
                    <td>Inside 20'</td>
                    <td>1</td>
                  </tr>
                  <tr>
                    <td>Inside 10'</td>
                    <td>2</td>
                  </tr>
                  <tr>
                    <td>Inside 2'</td>
                    <td>3</td>
                  </tr>
                </tbody>
              </Table>
            </div>
          </div>
        </Accordion.Body>
      </Accordion.Item>
      <Accordion.Item eventKey="3">
        <Accordion.Header>Events</Accordion.Header>
        <Accordion.Body>
          <div className="row justify-content-center">
            <div className="col-md-6">
              <Table bordered variant="dark" striped>
                <thead>
                  <tr>
                    <th>Event</th>
                    <th>Points</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>Participation</td>
                    <td>3</td>
                  </tr>
                  <tr>
                    <td>Par</td>
                    <td>1</td>
                  </tr>
                  <tr>
                    <td>Birdie</td>
                    <td>2</td>
                  </tr>
                  <tr>
                    <td>Eagle</td>
                    <td>4</td>
                  </tr>
                  <tr>
                    <td>Hole-In-One</td>
                    <td>10</td>
                  </tr>
                </tbody>
              </Table>
            </div>
          </div>
        </Accordion.Body>
      </Accordion.Item>
    </Accordion>
  );
};

export default PointsAccordion;
