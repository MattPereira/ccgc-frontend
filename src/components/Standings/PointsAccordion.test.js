import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import PointsAccordion from "./PointsAccordion";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <PointsAccordion />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <PointsAccordion />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});

it("displays the calculations accordion buttons", async function () {
  render(
    <MemoryRouter>
      <PointsAccordion />
    </MemoryRouter>
  );

  const strokesBtn = screen.getByText("Strokes");
  expect(strokesBtn).toBeInTheDocument();

  const puttsBtn = screen.getByText("Putts");
  expect(puttsBtn).toBeInTheDocument();

  const greeniesBtn = screen.getByText("Greenies");
  expect(greeniesBtn).toBeInTheDocument();

  const eventsBtn = screen.getByText("Events");
  expect(eventsBtn).toBeInTheDocument();
});
