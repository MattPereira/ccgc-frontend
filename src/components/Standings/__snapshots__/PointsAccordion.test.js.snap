// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-center accordion"
  >
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
      >
        <button
          aria-expanded="false"
          class="accordion-button collapsed"
          type="button"
        >
          Strokes
        </button>
      </h2>
      <div
        class="accordion-collapse collapse"
      >
        <div
          class="accordion-body"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-md-6"
            >
              <table
                class="table table-dark table-striped table-bordered"
              >
                <thead>
                  <tr>
                    <th>
                      Position
                    </th>
                    <th>
                      Points
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      1st
                    </td>
                    <td>
                      25
                    </td>
                  </tr>
                  <tr>
                    <td>
                      2nd
                    </td>
                    <td>
                      20
                    </td>
                  </tr>
                  <tr>
                    <td>
                      3rd
                    </td>
                    <td>
                      15
                    </td>
                  </tr>
                  <tr>
                    <td>
                      4th
                    </td>
                    <td>
                      10
                    </td>
                  </tr>
                  <tr>
                    <td>
                      5th
                    </td>
                    <td>
                      5
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
      >
        <button
          aria-expanded="false"
          class="accordion-button collapsed"
          type="button"
        >
          Putts
        </button>
      </h2>
      <div
        class="accordion-collapse collapse"
      >
        <div
          class="accordion-body"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-md-6"
            >
              <table
                class="table table-dark table-striped table-bordered"
              >
                <thead>
                  <tr>
                    <th>
                      Position
                    </th>
                    <th>
                      Points
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      1st
                    </td>
                    <td>
                      6
                    </td>
                  </tr>
                  <tr>
                    <td>
                      2nd
                    </td>
                    <td>
                      4
                    </td>
                  </tr>
                  <tr>
                    <td>
                      3rd
                    </td>
                    <td>
                      2
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
      >
        <button
          aria-expanded="false"
          class="accordion-button collapsed"
          type="button"
        >
          Greenies
        </button>
      </h2>
      <div
        class="accordion-collapse collapse"
      >
        <div
          class="accordion-body"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-md-6"
            >
              <table
                class="table table-dark table-striped table-bordered"
              >
                <thead>
                  <tr>
                    <th>
                      Event
                    </th>
                    <th>
                      Points
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      Greenie
                    </td>
                    <td>
                      1
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Inside 20'
                    </td>
                    <td>
                      1
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Inside 10'
                    </td>
                    <td>
                      2
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Inside 2'
                    </td>
                    <td>
                      3
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="accordion-item"
    >
      <h2
        class="accordion-header"
      >
        <button
          aria-expanded="false"
          class="accordion-button collapsed"
          type="button"
        >
          Events
        </button>
      </h2>
      <div
        class="accordion-collapse collapse"
      >
        <div
          class="accordion-body"
        >
          <div
            class="row justify-content-center"
          >
            <div
              class="col-md-6"
            >
              <table
                class="table table-dark table-striped table-bordered"
              >
                <thead>
                  <tr>
                    <th>
                      Event
                    </th>
                    <th>
                      Points
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      Participation
                    </td>
                    <td>
                      3
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Par
                    </td>
                    <td>
                      1
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Birdie
                    </td>
                    <td>
                      2
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Eagle
                    </td>
                    <td>
                      4
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Hole-In-One
                    </td>
                    <td>
                      10
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
